packages/aws-cdk-lib/aws-ses-actions/lib/s3.ts

```diff
@@ -55,7 +55,8 @@ export class S3 implements ses.IReceiptRuleAction {
       resources: [this.props.bucket.arnForObjects(`${keyPattern}*`)],
       conditions: {
         StringEquals: {
-          'aws:Referer': cdk.Aws.ACCOUNT_ID,
+          'aws:SourceAccount': cdk.Aws.ACCOUNT_ID,
+          'aws:SourceArn': `arn:${cdk.Aws.PARTITION}:ses:${cdk.Aws.REGION}:${cdk.Aws.ACCOUNT_ID}:receipt-rule-set/*:receipt-rule/*`,
```

Comment:
I think this is the closest we can get, given we don't have `rule_set_name` or `receipt_rule_name`.

Reply:
You have access to `rule.receiptRuleName`, is it not "the name of the receipt rule that contains the deliver to Amazon S3 bucket action"?

Reply:
Correct me if I'm wrong, but we're not getting a lot of additional restraints on the allocated perms with the `SourceArn` condition in its current state. The only thing we're gaining is the region restriction. The uncoditional `receipt-rule-set` doesn't help much, given we already only had the SES principal. I think trying to retrieve both the `ruleSet` from the `ReceiptRuleProps`, and the `receiptRuleName` (either generated, passed as a constructor prop, or imported), should be a priority

Reply:
Yeah, perhaps why this `Condition` is the way it is. May close this PR, I think `aws:Referer` with account ID is enough.